tools/scoring/tests/dimensions/entity_identification_test.py

```diff
@@ -17,34 +17,178 @@
+          matches=matches,
+          proposed_entities=deserialized_files_appended[PROPOSED],
+          solution_entities=deserialized_files_appended[SOLUTION])
+
+      return translations
+    elif entity_type == COMPLEX:
+      return deserialized_files_appended
+
   def setUp(self):
     super().setUp()
-    # TODO: add real data (append cases to existing tests)
-    deserialized_files = {PROPOSED: {}, SOLUTION: {}}
-    self.entity_identification = EntityIdentification(
-        deserialized_files=deserialized_files).evaluate()
-
-  def testDirectlyAssignedAttributes(self):
-    self.assertEqual(self.entity_identification.correct_reporting, 0)
-    self.assertEqual(self.entity_identification.correct_ceiling_reporting, 0)
-    self.assertEqual(self.entity_identification.incorrect_reporting, 0)
-
-  def testInheritedCalculatedAttributes(self):
-    self.assertEqual(self.entity_identification.correct_total(), 0)
-    self.assertEqual(self.entity_identification.correct_ceiling(), 0)
-    self.assertEqual(self.entity_identification.incorrect_total(), 0)
-
-  def testInheritedResultProperties(self):
-    # These are `None` by virtue of the ceiling being falsy.
-    self.assertEqual(self.entity_identification.result_all, None)
-    self.assertEqual(self.entity_identification.result_reporting, None)
-    self.assertEqual(self.entity_identification.result_virtual, None)
+    all_features = 'tests/samples/virtual_entity.yaml'
+    self.highest_score_argument = self._prepare_dimension_argument(
+        entity_type=COMPLEX,
+        proposed_path=all_features,
+        solution_path=all_features)
+
+    empty_file = 'tests/samples/empty.yaml'
+    self.none_score_argument = self._prepare_dimension_argument(
+        entity_type=COMPLEX, proposed_path=empty_file, solution_path=empty_file)
+
+  def testEntityIdentificationNoneScore(self):
+    """When ceiling==0, the resulting score is None."""
+    none_score = EntityIdentification(
+        deserialized_files=self.none_score_argument).evaluate()
+
+    # Directly assigned attributes
+    self.assertEqual(none_score.correct_reporting, 0)
+    self.assertEqual(none_score.correct_ceiling_reporting, 0)
+    self.assertEqual(none_score.incorrect_reporting, 0)
+
+    self.assertEqual(none_score.correct_virtual, 0)
+    self.assertEqual(none_score.correct_ceiling_virtual, 0)
+    self.assertEqual(none_score.incorrect_virtual, 0)
+
+    # Inherited calculated attributes
+    self.assertEqual(none_score.correct_total(), 0)
+    self.assertEqual(none_score.correct_ceiling(), 0)
+    self.assertEqual(none_score.incorrect_total(), 0)
+
+    # Inherited result properties. These are `None` by virtue of the ceiling
+    # being falsy: that is to say, there was nothing to score against.
+    self.assertEqual(none_score.result_all, None)
+    self.assertEqual(none_score.result_reporting, None)
+    self.assertEqual(none_score.result_virtual, None)
+
+  def testEntityIdentificationHighestScore(self):
```

Comment:
rename to 
`testEntityIdentificationWithSameReportingEntityHighestScore`

Reply:
I don't see how that adds clarity; the salient point is that there's 100% overlap. Perhaps "highest score" could be rephrased?

Reply:
That's correct. "Highest score possible," perhaps?